src/tools/nbench/nbench.cpp

```diff
@@ -298,8 +298,14 @@ OPTIONS
         return 1;
     }
 
-#ifdef NGRAPH_DISTRIBUTED
-    ngraph::Distributed dist;
+#if defined NGRAPH_DISTRIBUTED_ENABLE
+    auto dist = new ngraph::Distributed;
+    bool delete_instance = true;
+    if (dist->get_size() == 1)
+    {
+        delete dist;
+        delete_instance = false;
```

Comment:
I would suggest a small improvement - either use smart pointer e.g. unique_ptr or set dist to nullptr if distribution size is equal to 1, e.g.
```
if (dist->get_size() == 1)
{
    delete dist;
    dist = nullptr;
}
```
Later we could delete `dist` in condition `if(dist) delete dist;`

Reply:
done !